[IMG-175] Windows support policy (#8480)
circleci/circleci-docs: jekyll/_cci2/windows-images-support-policy.adoc

---
contentTags:
  platform:
  - Cloud
---
= CircleCI Windows images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI Windows images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

> This document outlines the CircleCI Windows image release, update, and deprecation policy. This policy applies to all Windows images built for the Windows Executor.

Reviewer: ```suggestion
This document outlines the CircleCI Windows image release, update, and deprecation policy. This policy applies to all Windows images built for the Windows executor.
```

[#release-policy]
== Release policy

The Windows images are based on Windows Server. We aim to maintain and release two major versions of Windows Server at all times. Currently we support `Windows server 2019` and `Windows server 2022`

> Tags will be redirected to the updated images automatically as described in the xref:#tagging[tagging] section of this document. We will announce these releases on our link:https://discuss.circleci.com/[Discuss Forum].

Reviewer: ```suggestion
Tags will be redirected to the updated images automatically as described in the <<tagging>> section of this document. We will announce these releases on our link:https://discuss.circleci.com/[Discuss Forum].
```

[#tagging]
== Tagging

We support various tags for the Windows environment, allowing you to choose a suitable image for your needs.

- `default`: version is what you will receive if do not specify an image.

> - `current`: version of Windows Server will receive updates approximately every 3 months. This tag will be the same version as what is provided if no tag is provided.

Reviewer: ```suggestion
- `current`: version of Windows Server will receive updates approximately every three months. This tag will be the same version as what is provided if no tag is specified.
```

> - `edge`: tags are reserved for previews of new releases, which will initially point to this tag. The `edge` tags may include incremental updates to the current quarterly image release, which may change without notice, and is not recommended to be used for production CI workloads. `current` will be updated with these changes after period of stability.

Reviewer: ```suggestion
- `edge`: tags are reserved for previews of new releases, which will initially point to this tag. The `edge` tags may include incremental updates to the current quarterly image release, which may change without notice, and is not recommended to be used for production CI workloads. `current` will be updated with these changes after a period of stability.
```

- `previous`: version of Windows Server will receive the previous version of `current`.

We generally install the latest Major version of Visual Studio Community Edition that coincides with the Windows Server version in the image when it is built.

> We always aim to release a version of the supported windows images (e.g. currently this is `Windows server 2019` and `Windows server 2022`) every 3 months. We recommend using the `default` version and not pinning to a date version.

Reviewer: ```suggestion
We always aim to release a version of the supported windows images (for example, currently this is `Windows server 2019` and `Windows server 2022`) every 3 months. We recommend using the `default` version and not pinning to a date version.
```

> == Tagging with the Windows Orb:

Reviewer: ```suggestion
[#tagging-with-the-windows-orb]
== Tagging with the Windows orb
```

> The variant of Windows Server can be specified by choosing the name parameter in the Windows Orb. The most recent version of Windows Server will always be accessible via the `win/default` name, but previous versions require explicit definition, such as `win/server-2019`.

Reviewer: ```suggestion
The variant of Windows Server can be specified by choosing the name parameter in the Windows orb. The most recent version of Windows Server will always be accessible via the `win/default` name, but previous versions require explicit definition, such as `win/server-2019`.
```

> Each quarterly (published every 3 months) image for the current tag Windows Server version is published to a date-based tag, such as 2023.10.1, which can be explicitly defined in the version parameter of the Orb.

Reviewer: ```suggestion
Each quarterly (published every 3 months) image for the `current` tag Windows Server version is published to a date-based tag, such as `2023.10.1`, which can be explicitly defined in the version parameter of the orb.
```

[#critical-cve-patches]
== Critical CVE patches

> When critical CVEs are disclosed that affect the versions of the operating system or software stack in our Windows Images, we will investigate the impact that this has on our images being used within the CircleCI execution environment. If customers are impacted by these CVEs we will push a patch fix to the released image(s), this image will supersede the original image.

Reviewer: ```suggestion
When critical CVEs are disclosed that affect the versions of the operating system or software stack in our Windows images, we will investigate the impact that this has on our images being used within the CircleCI execution environment. If customers are impacted by these CVEs we will push a patch fix to the released image(s). This image will supersede the original image.
```

[#bug-reports-issues-and-prs]
== Bug reports, issues, and PRs

You can file a link:https://support.circleci.com/hc/en-us/requests/new[support ticket] with CircleCI Support for any issues or bugs found with the Windows images. Our support team will be able to escalate issues internally to the correct engineering team and provide updates on the issue.

[#image-lifespan-eol]
== Image lifespan / EOL

> Once the next major version of Windows Server is released by Microsoft and generally available to customers, we will deprecate the oldest version of Windows Server that we support and schedule it to be removed.

Reviewer: ```suggestion
Once the next major version of Windows Server is released by Microsoft, and generally available to customers, we will deprecate the oldest version of Windows Server that we support and schedule it to be removed.
```

> After the new version of Windows Server has been released, we will then move the current Windows Server version into an extended support phase and move to the tagging system described in the Tagging section of this document.

Reviewer: ```suggestion
After the new version of Windows Server has been released, we will then move the current Windows Server version into an extended support phase and move to the tagging system described in the <<tagging>> section of this document.
```

> Current Deprecation (Jan 2024):

Reviewer: ```suggestion
Current deprecation (Jan 2024):
```

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Windows Server 2019
| Build quarterly images and retain Q4 (October) releases from previous two years

| Windows Server 2022

> | Build quarterly images and Q4 (October) releases retained

Reviewer 2: We wanted specific wording for this to show we will only keep Q4 image each year: `Quarterly images built and only Q4 image retained each year`

> | Build quarterly images and only Q4 (October) releases retained each year

Reviewer 2: Please remove build

Author: This is what is on the policy doc

Reviewer 2: Windows Server 2019 | Retain only Q4 (October) releases (total of 2 images: one image each from 2021, 2022)Quarterly images built and only Q4 image retained each year
-- | --

Reviewer 2: the general wording has changed. Am ok with this for this one, but moving forward we need to keep something like this where we have approval to just copy paste

|===

> Example: When Windows Server 2025 is Released

Reviewer: ```suggestion
Example: When Windows Server 2025 is released
```

[.table.table-striped]
[cols=2*, options="header", stripes=even]
|===
| Version
| Support

| Windows Server 2019
| Deprecated and removed entirely

| Windows Server 2022
| Retain only current and past two Q4 (October) releases

| Windows Server 2025

> | Quarterly images built and only Q4 (October) image retained

Reviewer 2: `Quarterly images built and only October (Q4) image retained each year`

|===

When an image is selected for deprecation and removal, we will create an announcement on our Discuss forum, along with reaching out via email to developers who have requested one of the deprecated images in their recent jobs.

> We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release

Reviewer: ```suggestion
We will also plan brownouts to ensure users are aware of the approaching removal of deprecated images. Generally, we will aim to start an EOL process within 3 months of a new version release.
```

---
contentTags:
  platform:
  - Cloud

>   - Server v4.x

Reviewer 2: Same question about server?

  - Server v3.x
---
= CircleCI Windows images support policy
:page-layout: classic-docs
:page-liquid:
:page-description: CircleCI Windows images release, update, and deprecation policy
:icons: font
:experimental:

[#overview]
== Overview

This document outlines the CircleCI Windows image release, update, and deprecation policy. This policy applies to all Windows images built for the Windows Executor.

[#release-policy]
== Release policy

The Windows images are based on Windows Server. We aim to maintain and release two major versions of Windows Server at all times. Currently we support `Windows server 2019` and `Windows server 2022`

Tags will be redirected to the updated images automatically as described in the xref:#tagging[tagging] section of this document. We will announce these releases on our link:https://discuss.circleci.com/[Discuss Forum].

[#tagging]
== Tagging

We support various tags for the Windows environment, allowing you to choose a suitable image for your needs.

> - `default`: This image is the current stable image. Jobs will default to this if no image is specified.

Reviewer 2: version is what you will receive if do not specify an image.

> - `current`: This version of Windows will receive updates approximately every 3 months. If no tag is provided, `current` will be the version that jobs default to.

Author: will do

> - `edge`: This tag is reserved for previews of new releases, which will initially point to this tag. The tag may include incremental updates relative to the current quarterly image release, which may change without notice, and is not recommended for production CI workloads. The `current` tag is promoted from `edge`

Reviewer 2: tags are reserved for previews of new releases, which will initially point to this tag. The edge tags may include incremental updates to the current quarterly image release, which may change without notice, and is not recommended to be used for production CI workloads. currentwill be updated with these changes after period of stability.

Author: will do

> - `previous`: Once an `edge` image is promoted to `default` and `current`, the previous `default` and `current` image is moved to the `previous` tag.

Reviewer 2: previous version of Windows Server will receive the previous version of current.

Author: will do
